rest_witchcraft/fields.py

```diff
@@ -5,20 +5,37 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, print_function, unicode_literals
 
-from rest_framework import fields
+from rest_framework import fields, relations
 
+from django_sorcery.db.meta import model_info
 
-# TODO: Should build a url to the resource
 
+class HyperlinkedIdentityField(relations.HyperlinkedIdentityField):
+    def get_url(self, obj, view_name, request, format):
+        info = model_info(obj.__class__)
 
-class UriField(fields.CharField):
+        # Unsaved objects will not yet have a valid URL.
+        if not all(getattr(obj, i) for i in info.primary_keys):
```

Comment:
https://github.com/shosca/django-sorcery/blob/master/django_sorcery/db/models.py#L38

Reply:
that just returns the first pk. im checking all here